test/smoke/gateway_single_spoke_test.go

```diff
@@ -0,0 +1,267 @@
+package smoke
```

Comment:
Why "smoke" and not "e2e"? What's the intention of these tests, to execute and e2e suite with different scenarios in a test cluster only, or run smoke tests against any cluster in a non destructive way to check the cluster is working?

Reply:
The original intent was to run them against any cluster, when the plan was having a unstable and stable clusters running the service. When that changed I reworked the tests to work against the local-setup and never changed the name. But now that you mention it, it probably doesn't make sense anymore and we might just call them e2e.

Reply:
We should probably just rename the directory and package to `e2e` i.e. `test/e2e` ?